fx/client_test.go

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package dosafx_test
+
+import (
+	"testing"
+
+	fxconfig "go.uber.org/fx/config"
+	"go.uber.org/fx/service"
+	"go.uber.org/yarpc"
+	"go.uber.org/yarpc/transport/tchannel"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/uber-go/dosa"
+	"github.com/uber-go/dosa/config"
+	_ "github.com/uber-go/dosa/connectors/devnull"
+	"github.com/uber-go/dosa/fx"
+)
+
+type InvalidEntity struct {
+	dosa.Entity `dosa:"primaryKey="`
+	PrimaryKey  int64
+	data        string
+}
+
+type ValidEntity struct {
+	dosa.Entity `dosa:"primaryKey=PrimaryKey"`
+	PrimaryKey  int64
+	data        string
+}
+
+func getTestDispatcher(t *testing.T) *yarpc.Dispatcher {
+	ts, _ := tchannel.NewChannelTransport(tchannel.ServiceName("dosatest"))
+	ycfg := yarpc.Config{
+		Name: "service",
+		Outbounds: yarpc.Outbounds{
+			"dosa-dev-gateway": {
+				Unary: ts.NewSingleOutbound("127.0.0.1:21300"),
+			},
+		},
+	}
+	return yarpc.NewDispatcher(ycfg)
+}
+
+// TODO(eculver): these tests need some love, particularly in mocking out the
```

Comment:
Suggestion: add a config that uses the null connector and test that way. That will at least test all the client code without testing the configuration of yarpc.